File: go/regressions/uimd/parity/source_separator_scroll/source_separator_scroll.go
package main

import (
	"os"

	uimd "uimd"
)

type SourceSeparatorScroll struct {
	*SourceSeparatorScrollUI
	Panel *uimd.ScrollView
}

func NewSourceSeparatorScroll() *SourceSeparatorScroll {
	app := &SourceSeparatorScroll{SourceSeparatorScrollUI: NewSourceSeparatorScrollUI()}
	app.Panel = app.NewGeneratedScrollViewElement()
	app.AddElement(app.Panel)
	app.Panel.ClearChildren()
	row := NewSourceSeparatorRowUI()
	reusable := uimd.NewReusableElement("row", "source_separator_row")
	reusable.SetChild(row)
	naturalSize := uimd.GeneratedWindowContentSize(row)
	reusable.SetFrame(uimd.Rect{Width: naturalSize.Width, Height: naturalSize.Height})
	app.Panel.AddChild(reusable)
	return app
}

func main() {
	code := uimd.RunGeneratedAppMain(func() int {
		app := NewSourceSeparatorScroll()
		return uimd.RunGeneratedWindow(app, app.RuntimeOptions(), os.Args)
	})
	os.Exit(code)
}

File: java/regressions/uimd/parity/source_separator_scroll/SourceSeparatorScroll.java
import uimd.GeneratedWindowRuntime;
import uimd.Rect;
import uimd.ReusableElement;
import uimd.Size;

public final class SourceSeparatorScroll extends SourceSeparatorScrollUI
{
    public SourceSeparatorScroll()
    {
        clearChildren();
        SourceSeparatorRowUI row = new SourceSeparatorRowUI();
        row.setMode("expand_width");
        Size natural = GeneratedWindowRuntime.generatedWindowContentSize(row);
        ReusableElement reusable = new ReusableElement("row", row);
        reusable.setFrame(new Rect(0, 0, natural.width(), natural.height()));
        addChild(reusable);
    }

    public static void main(String[] arguments)
    {
        System.exit(GeneratedWindowRuntime.runGeneratedWindow(
            new SourceSeparatorScroll(),
            arguments));
    }
}
